File: Programmers/pro_42587_프린터.py
import heapq
from collections import deque

def solution(priorities, location):
    max_heap = []
    idx_prio = deque()
    for idx, num in enumerate(priorities):
        heapq.heappush(max_heap,-num)
        idx_prio.append((num,idx))
    
    cnt=0
    while(True):
        top = idx_prio.popleft()
        if(top[0]== -1*max_heap[0]):
            cnt+=1
            if(top[1]==location):
                break
            heapq.heappop(max_heap)
        else:
            idx_prio.append(top)         
    return cnt

File: Programmers/pro_42587_프린터.js
function solution(priorities, location) {
    var answer = 0;
    const prio = priorities.map((v, i) => [v, i])
    while (prio.length) {
        let [top, idx] = prio.shift()
        let flag = 0
        for (let i = 0; i < prio.length; i++) {
            if (top < prio[i][0]) {
                flag = 1
                break
            }
        }
        if (flag) {
            prio.push([top, idx])
        } else {
            answer += 1
            if (idx == location)
                return answer
        }
    }
    return -1;
}